### PR TITLE
Fix crash in lowering use of a global variable

### PR DESCRIPTION
**Pull request by dwblaikie**
Actually presenting two options in this one review - if you look at the specific
commits in this PR, the first commit represents my first attempt - and if you
look at the overall PR change for the second attempt.

But I'm totally open to completely different approaches/ideas - these were just
my rough guesses.

### CODE REVIEW

File: toolchain/lower/testdata/global/fun_use.carbon [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/global/fun_use.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/global/fun_use.carbon
```
Comment (josh11b) [inline toolchain/lower/testdata/global/fun_use.carbon]: We don't abbreviate "function" as "fun", https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/docs/idioms.md#abbreviations-used-in-the-code-aka-carbon-abbreviation-decoder-ring ; use either "function" or "fn"

Comment (dwblaikie) [reply toolchain/lower/testdata/global/fun_use.carbon]: Ah, right - thanks!
Renamed to just `use.carbon` - and added a use of one global in the initialization of another global, for some variety/broader testing.

File: toolchain/lower/testdata/global/use.carbon [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/global/use.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/global/use.carbon
+var a: i32;
+
+var b: i32 = a;
```
Comment (josh11b) [inline toolchain/lower/testdata/global/use.carbon]: I would imagine a future version of the compiler rejecting this code, for using a variable in unformed state. Better to keep that out of a test that isn't trying to test that thing.

Comment (dwblaikie) [reply toolchain/lower/testdata/global/use.carbon]: Ah, I wanted to test the use of the global from another global. If I initialized `a` with some value, would that be acceptable/less likely to be made invalid in the future?

Comment (josh11b) [reply toolchain/lower/testdata/global/use.carbon]: Yes

Comment (dwblaikie) [reply toolchain/lower/testdata/global/use.carbon]: Done in 150beab9fa18b199bfe27061c64e2225135b01b4

**Review by josh11b — APPROVED** [review]

**Pull request merged**